### PR TITLE
Setup CI

### LINKED ISSUE #2: Setup CI
Once the code is copied over to this repository (#1 ), we should be able to setup CI in this repository. Travis and Appveyor should both be setup such that they check if the code is flake8 clean and if the testsuite passes.

### PR DESCRIPTION
**Pull request by aaronayres35**
Fixes #2 

Note this PR sits on top of #4.  I have this PR attempting to merge into the branch for #4 to keep the files changed specific to this PR.

This PR adds an `etstool.py` file (I followed the example of `traitsui` and `pyface`, but it may be better to have a `ci` directory more like `traits-futures`? see https://github.com/enthought/traits-futures/tree/master/ci)
It also adds `.travis.yml` and `appveyor.yml` files which run `flake8` and the test suite.

Also, the commit history is rather ugly (the only way I know of to debug this CI code is to make a new commit and push to run CI), so remember to squash merge!

### CODE REVIEW

**Comment by aaronayres35** [comment]
Because I am not merging to master appveyor is not running.  I am changing this PR to base on master so appveyor CI can run

**Comment by aaronayres35** [comment]
> Because I am not merging to master appveyor is not running. I am changing this PR to base on master so appveyor CI can run

I thought this was the problem (appveyor wasn't even being listed as a check previously).  However, now appveyor still isn't running and is stuck on "Expected — Waiting for status to be reported".  I will investigate more tomorrow morning.

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  matrix:
+    - RUNTIME: '3.6'
+
+matrix:
+  fast_finish: true
+
+branches:
+  only:
+    - master
```
Comment (kitchoi) [inline appveyor.yml]: If we remove this block then persumably we can run CI even if this PR does not target master.

Comment (aaronayres35) [reply appveyor.yml]: I removed it.  Should I revert back to have this PR target the branch for #4 now?

**Comment by rahulporuri** [comment]
I manually set the "Default branch" to "setup-ci" (this branch) in the appveyor settings. I think once this PR is merged, I'll need to update the branch name back to master.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+language: generic
+dist: xenial
+services:
+  - xvfb
```
Comment (rahulporuri) [inline .travis.yml]: don't think we need to setup the `xvfb` service for the copyright checker.

**Review by rahulporuri — APPROVED** [review]
mostly LGTM

**Comment by rahulporuri** [comment]
> This PR adds an etstool.py file (I followed the example of traitsui and pyface, but it may be better to have a ci directory more like traits-futures? see https://github.com/enthought/traits-futures/tree/master/ci)
It also adds .travis.yml and appveyor.yml files which run flake8 and the test suite.

That might be overkill. In fact, i think using the `etstool.py` file itself is maybe overkill.
Another option is to simply do what traits does - not rely on installing edm and `etstool.py`.

**Comment by aaronayres35** [comment]
After another commit was pushed it looks like appveyor did not run / was not required to run

**Comment by rahulporuri** [comment]
@aaronayres35 i manually triggered appveyor again so it runs. not sure what the right fix here is :/

**Comment by rahulporuri** [comment]
looks like CI is happy. please squash merge.

**Comment by aaronayres35** [comment]
> That might be overkill. In fact, i think using the `etstool.py` file itself is maybe overkill.
> Another option is to simply do what traits does - not rely on installing edm and `etstool.py`.

~I see, should I remove the `etstool.py` from this PR and instead follow the traits approach, or should we leave as is for now and possibly shift to the traits model later?~
EDIT: Just seeing your latest comment.  I will squash merge as is

**Comment by rahulporuri** [comment]
> I see, should I remove the etstool.py from this PR and instead follow the traits approach, or should we leave as is for now and possibly shift to the traits model later?

I would prefer later. i prefer pushing this through now and actually using it - after which we can tweak it/clean it up.

**Pull request merged**